lib/ansible/modules/cloud/hcloud/hcloud_network.py

```diff
@@ -0,0 +1,219 @@
+
+
+class AnsibleHcloudNetwork(Hcloud):
+    def __init__(self, module):
+        Hcloud.__init__(self, module, "hcloud_network")
+        self.hcloud_network = None
+
+    def _prepare_result(self):
+        return {
+            "id": to_native(self.hcloud_network.id),
+            "name": to_native(self.hcloud_network.name),
+            "ip_range": to_native(self.hcloud_network.ip_range),
+            "labels": self.hcloud_network.labels,
+        }
+
+    def _get_network(self):
+        try:
+            if self.module.params.get("id") is not None:
+                self.hcloud_network = self.client.networks.get_by_id(
+                    self.module.params.get("id")
+                )
+            else:
+                self.hcloud_network = self.client.networks.get_by_name(
+                    self.module.params.get("name")
+                )
+        except APIException as e:
+            self.module.fail_json(msg=e.message)
+
+    def _create_network(self):
+
+        self.module.fail_on_missing_params(
+            required_params=["name", "ip_range"]
+        )
+        params = {
+            "name": self.module.params.get("name"),
+            "ip_range": self.module.params.get("ip_range"),
+            "labels": self.module.params.get("labels"),
+        }
+
+        if not self.module.check_mode:
+            self.client.networks.create(**params)
+
+        self._mark_as_changed()
+        self._get_network()
+
+    def _update_network(self):
+
+        labels = self.module.params.get("labels")
+        if labels is not None and labels != self.hcloud_network.labels:
+            if not self.module.check_mode:
+                self.hcloud_network.update(labels=labels)
+            self._mark_as_changed()
+
+        self._get_network()
+
+    def present_network(self):
+        self._get_network()
+        if self.hcloud_network is None:
+            self._create_network()
+        else:
+            self._update_network()
+
+    def delete_network(self):
+        self._get_network()
+        if self.hcloud_network is not None:
+            if not self.module.check_mode:
+                self.client.networks.delete(self.hcloud_network)
+            self._mark_as_changed()
+        self.hcloud_network = None
+
+    @staticmethod
+    def define_module():
+        return AnsibleModule(
+            argument_spec=dict(
+                id={"type": "int"},
+                name={"type": "str"},
+                ip_range={"type": "str"},
+                labels={"type": "dict"},
+                state={
+                    "choices": ["absent", "present"],
+                    "default": "present",
+                },
+                **Hcloud.base_module_arguments()
+            ),
+            required_one_of=[['id', 'name']],
+            supports_check_mode=True,
+        )
+
+
+def main():
+    module = AnsibleHcloudNetwork.define_module()
+
+    hcloud = AnsibleHcloudNetwork(module)
+    state = module.params.get("state")
```

Comment:
You can use `[]` instead of `.get()` for options. Makes it easier to read IMO. Feel free to change all occurrences or to ignore this, whatever you prefer :)
```suggestion
    state = module.params["state"]
```